midterm-2023-11-06/code/5215-berat-useini/script.ts

```diff
@@ -0,0 +1,191 @@
+type RockBand = {
+    ID: number;
+    name: string;
+    formed: number;
+    location: string;
+    country: string;
+    members: string;
+    albums: number;
+};
+
+interface albums{ 
+name: string;
+
+}
+
+document.addEventListener("DOMContentLoaded", siteCode);
+let rockband : RockBand[] = [];
+async function siteCode() {
+    const data = await loadData();
+    rockband = data.metalBands;
+
+    fillLocation(rockband)
+
+    const sortButton = document.getElementById("sort-id")!
+    sortButton.addEventListener("click", sortById)
+
+
+    const sortName = document.getElementById("sort-name")!
+    sortName.addEventListener("click", sortByName)
+
+    const applyName = document.getElementById("name-filter")
+    applyName?.addEventListener("input", applyNameFilter)
+
+ 
+
+    const applyButton = document.getElementById("apply-filter")!
+    applyButton.addEventListener("click", applyFilter)
+    populateTable(rockband)
+
+
+}
+
+// Function to populate the HTML table with rock band data
+
+const fillLocation = (rockband: RockBand[]) => {
+    const filter = document.getElementById("country-filter")!;
+
+    const countries = new Set<string>();
+    for (const bandrock of rockband) {
+        countries.add(bandrock.country);
+    }
+
+    for (const country of countries) {
+        const option = document.createElement("option");
+        option.value = country;
+        option.innerHTML = country;
+        filter.appendChild(option);
+    }
+}
+
+
+const applyFilter = () => {
+    const CountriesElement = document.getElementById("country-filter") as HTMLSelectElement;
+    const Countries = CountriesElement.value;
+
+    
+
+    let filteredCountries= rockband;
+    if (Countries!== "all") {
+       filteredCountries= filteredCountries.filter(band=> band.country === Countries);
+    }
+
+    populateTable(filteredCountries);
+
+}
+
+
+
+const loadData = async () => {
+    const dataUri = 'https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json';
+    const response = await fetch(dataUri);
+    if (!response.ok) {
+        throw new Error("The data is not there");
+    }
+    const data = await response.json();
+    return data;
+
+
+
+}
+
+type BandSorter = (first: RockBand, second: RockBand) => number;
+
+const nameSorter: BandSorter = (first, second) => first.name.localeCompare(second.name);
+const idSorter: BandSorter = (first, second) => first.ID - second.ID;
+
+const sortByName = () => {
+    const sortedBands = rockband.slice().sort(nameSorter);
+    populateTable(sortedBands);
+}
+
+// Function to apply the name filter
+const applyNameFilter = () => {
+    const nameFilterInput = document.getElementById("name-filter") as HTMLInputElement;
```

Comment:
This will successfully filter the data by the name of the band. +10
Kudos on using the `toLowerCase` function +5